test/floodsub.spec.js

```diff
@@ -4,107 +4,113 @@
 const { expect } = require('aegir/utils/chai')
 const sinon = require('sinon')
 const uint8ArrayFromString = require('uint8arrays/from-string')
-
+const uint8ArrayToString = require('uint8arrays/to-string')
+const { sha256 } = require('multiformats/hashes/sha2')
 const { utils } = require('libp2p-interfaces/src/pubsub')
-const pWaitFor = require('p-wait-for')
+const { SignaturePolicy } = require('libp2p-interfaces/src/pubsub/signature-policy')
+const PeerStreams = require('libp2p-interfaces/src/pubsub/peer-streams')
+const PeerId = require('peer-id')
 
 const Floodsub = require('../src')
-const { createPeers } = require('./utils/create-peer')
 
 const defOptions = {
-  emitSelf: true
+  emitSelf: true,
+  globalSignaturePolicy: SignaturePolicy.StrictNoSign
 }
 
 const topic = 'my-topic'
 const message = uint8ArrayFromString('a neat message')
 
 describe('floodsub', () => {
-  let floodsub1, floodsub2
-  let peer1, peer2
+  let floodsub
 
   before(async () => {
     expect(Floodsub.multicodec).to.exist()
 
-    ;[peer1, peer2] = await createPeers({ number: 2 })
-    floodsub1 = new Floodsub(peer1, defOptions)
-    floodsub2 = new Floodsub(peer2, defOptions)
+    const libp2p = {
+      peerId: await PeerId.create(),
```

Comment:
any reason to not use createPeer?

Reply:
Spinning up libp2p nodes is overkill for unit tests.  Ideally we wouldn't depend on libp2p at all in this module as libp2p itself depends on floodsub creating a loop which we really need to break in order to simplify publishing new versions.